### PR TITLE
Make Tweaks

### PR DESCRIPTION
**Pull request by Fenikkusu**
This PR:

- [x] Adds new feature ...
- [ ] Covered by tests
- [ ] Doc PR: https://github.com/infection/site/pull/XXX

Fixes https://github.com/infection/infection/issues/318

I made a few tweaks to the make process, just to make it a little easier to work with phars locally.

1. Added a target for the .travis/infection-private.pem - I presume travis will add this automatically during a normal run, if it is so needed.
1. Added a `link` target - just a quick helper to link to /usr/local/bin/infection
1. Added a `clean` target - removes a previously created phar and link.
1. Updated .gitignore - If you run box.phar multiple times (IE: -B make flag), you phar file will become box.phar.1, box.phar.2, etc.

### CODE REVIEW

**Comment by theofidry** [comment]
As per https://github.com/infection/infection/issues/318#issuecomment-385000490, I don't think we should require the signing algorithm for the dev used only for dev purposes. But for the real PHAR (which can be build and used in the CI and then deployed automatically), I think it would be better if @borNfreee would do it as described in [create and deploy secure PHARs](https://medium.com/@tfidry/create-and-deploy-secure-phars-c5572f10b4dd). I think however it should be a GPG key for the Infection project (there's a link about that in the article as well)

File: Makefile [hunk 1]
```diff
@@ -125,9 +125,22 @@ phpstan: vendor $(PHPSTAN)
...
+	openssl req -newkey rsa:2048 -new -nodes -x509 -days 3650 -keyout .travis/infection-private.pem -out /dev/null -subj "/C=GB/ST=London/L=London/O=Global Security/OU=IT Department/CN=example.com"
+
+.PHONY: clean
+clean:
+	rm -rf box.phar*
+	rm -rf build/bin/infection.phar
+	rm -rf /usr/local/bin/infection
+
+.PHONY: link
+link: build/bin/infection.phar
```
Comment (theofidry) [inline Makefile]: Hm you're sure about that?

**Comment by Fenikkusu** [comment]
I do have to agree that signing for local development may not be necessary.

File: Makefile (hunk 1 above)
Comment (Fenikkusu) [reply Makefile]: Sure about the linking? I just see it as a quick helper for development. Probably wouldn't do anything for windows users.

**Comment by borNfreee** [comment]
this is not relevant anymore, use `make build/bin/infection.phar` to build not signed local PHAR for development/test

**Pull request closed without merging**